File: Python/Array_101/4_Searching/mountain_array.py
class Solution:
    def validMountainArray(self, arr: List[int]) -> bool:
        n = len(arr)
        if n < 3:
            return False
        up = False
        down = False
        for i in range(1, n):
            if arr[i-1] == arr[i]:
                return False
            if up == 0:
                if arr[i-1] < arr[i]:
                    up += 1
                else:
                    return False
            elif down == 0:
                if arr[i-1] > arr[i]:
                    down += 1
            else:
                if arr[i-1] < arr[i]:
                    return False
        if down != 1:
            return False
        return True

File: Java/Array_101/4_Searching/MountainArray.java
public class MountainArray {
    class Solution {
        public boolean validMountainArray(int[] arr) {
            int n, l, r;
            n = arr.length;
            l = 0;
            r = n - 1;
            while (l < n - 1) {
                if (arr[l] >= arr[l + 1])
                    break;
                l++;
            }
            if (l == 0 || l == n - 1)
                return false;
            while (r > 0) {
                if (arr[r] >= arr[r - 1])
                    break;
                r--;
            }
            return l == r;
        }
    }
}
